projects/components/src/download-file/download-file.component.test.ts

```diff
@@ -19,36 +21,32 @@ describe('Download File Component', () => {
     fileName: 'download.txt'
   };
 
-  test('should have loading icon, when data is loading', () => {
+  test('should download file', fakeAsync(() => {
     const spectator = createComponent({
       props: {
         metadata: metadata
       },
       providers: [
         mockProvider(FileDownloadService, {
-          downloadAsText: jest.fn(),
-          fileDownloadEvent$: of(FileDownloadEventType.Progress)
+          downloadAsText: jest.fn().mockReturnValue(timer(2000))
         })
       ]
     });
 
+    // Before data loading
+    expect(spectator.query(ButtonComponent)).toExist();
+    expect(spectator.query(IconComponent)).not.toExist();
+
+    // While data loading
+    spectator.click('.download-file');
     expect(spectator.query(ButtonComponent)).not.toExist();
     expect(spectator.query(IconComponent)).toExist();
-  });
+    spectator.tick(2500);
```

Comment:
Can we keep a smaller tick? We can reduce the timer as well. This will delay the tests unneccessarily

Reply:
Didn't review the PR, but just want to point out this isn't a sleep, it's fake async - won't change test runtime.

Reply:
Okay